### ISSUE TITLE
How to update the trait value 

### ISSUE THREAD

**Issue by abzal0** [artf/grapesjs#2628]
Hello @artf  thanks a lot for a great project, 
I have created a new type, it is similar to regular link, but the link title updates with ajax. Then updates the trait 'linktitle'. The value of it and the canvas do update, but the trait inside "Settings" still shows the initial value and does not changes. 
Part of the code that must update the value:

editor.DomComponents.addType('newLink', {
model: {
    defaults: {
         components: [	
				{
					highlightable: true,
					editable: true,
					selectable: true
				}],
         traits: [
						{
							type: 'text',
							label: 'Url',
							name: 'url',
							value: 'https://',
							changeProp: 0
						},
						{
							type: 'text',
							label: 'Title',
							name: 'linktitle',
							value: 'Link title', //this must be updated
						changeProp: 0
						}
      ]}},
        view:{
          		tagName: 'div',
			init({ model }) {
			this.listenTo(model, 'change:attributes', model => {
	//First attempt	
editor.getSelected().updateTrait('linktitle', {
type: 'text',
label: 'Title 2',
name: 'linktitle',
value: 'AA', //updating the value to AA
changeProp: 0
});
//shows undefined on canvas but only updates the label, need to update the value
//second attempt
this.model.attributes.attributes.title=' BB ';	
//updates the canvas link to BB, but does not update the trait
editor.trigger('component:toggled');
editor.trigger('component:updated');
editor.trigger('component:update:linktitle');
editor.trigger();  
				});
			},
        }
   }
)

Any suggestion how to update the trait value would be helpful.

**Comment by abzal0** [comment]
editor.getSelected().getTrait('linktitle').set('value','CCC');

does not works too

**Comment by pouyamiralayi** [comment]
@abzal0  you must bind your trait to component prop by setting ``changeProp:true``.
Cheers!

**Comment by abzal0** [comment]
@pouyamiralayi thank you for your suggestion, but as you can see on the code 'linktitle' already has 'changeProp: 0'. 
The value of 'linktitle' does updates, the link appearance does updates with value on a canvas, but the value inside of the input text under Settings tab has the same old value. It even updates the label next to input text box, but not the value inside.

**Comment by pouyamiralayi** [comment]
@abzal0 ``changeProp: 0``  means setting it to ``false``!

**Comment by abzal0** [comment]
@pouyamiralayi thank you for clarifying that, I changed it to 1. When I dropped it into canvas, under Settings, link title input text box does not has a predefined value of 'Link title', when I change changeProp to 0 it has the value:
![image](https://user-images.githubusercontent.com/5952350/76209828-408fe580-621c-11ea-8037-ee742960b3ca.png)

Updating its value still does not works.

**Comment by pouyamiralayi** [comment]
@abzal0 inside the [init](https://grapesjs.com/docs/modules/Components.html#lifecycle-hooks) hook of your component, update the value of your prop:
```
model: {
    defaults: {
      testprop: 1,
    },
    init() {
      // Here you can issue an ajax call and fill the prop with the response
    },
```
Cheers!

**Comment by abzal0** [comment]
@pouyamiralayi thank you for explaining in details, it updates on canvas but not on Settings tab. 


model: {
    defaults: {
      linktitle: 'Link title',
    },
    init() {
      // Here you can issue an ajax call and fill the prop with the response
 this.model.defaults.linktitle=' hello '; // changing the value
    },


Result:
Canvas:
![image](https://user-images.githubusercontent.com/5952350/76212716-89e33380-6222-11ea-8cd4-a944d55692f7.png)

Style manager Settings tab input text still has default value:
![image](https://user-images.githubusercontent.com/5952350/76212752-9c5d6d00-6222-11ea-907c-07d7cc64ca8e.png)

**Comment by pouyamiralayi** [comment]
@abzal0 alright! although the prop approach must work fine, i am not sure why it is not, or i am missing something here! if we want to go with attributes, you must use ``addAttributes`` for changing the attribute, here is a working example of the procedure:
```
  domc.addType('test', {
        model: {
            init() {
                this.listenTo(this, 'change:attributes:linkTitle', () => console.log('linkTitle', this.getAttributes()['linkTitle']))
            },
            defaults: {
                attributes: {
                    linkTitle: '',
                },
                style: {
                    padding: '25px',
                    margin: '25px',
                },
                traits: [
                    {type: 'text', name: 'linkTitle', changeProp: false, label: 'Link Title'}
                ]
            }
        },
        view: {
            init() {
                const oldValue = this.model.getAttributes()['linkTitle']
                console.log('linkTitle: ', oldValue)
                this.model.addAttributes({'linkTitle':oldValue+' Changed!'})
            }
        }
    })
```
Cheers!

**Comment by abzal0** [comment]
@pouyamiralayi thanks a lot for your time, this solved the issue:
`                this.model.addAttributes({'linkTitle':oldValue + ' Changed!'})
`
you saved so much time, thanks again :)
